### PR TITLE
Push demo artifact to docker hub

### PR DESCRIPTION
**Pull request by alpe**
Update circleCI build steps to build and push `wasmd-demo` artifact
______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -195,20 +195,18 @@ jobs:
           at: /tmp/workspace
       - checkout
       - setup_remote_docker:
-          docker_layer_caching: true
-      - run: |
-          GAIAD_VERSION=''
+          docker_layer_caching: yes
+      - run:
+          name: Build Docker artifact - demo
+          command: docker build --pull -t "cosmwasm/wasmd-demo-demo:${CIRCLE_SHA1}" .
```
Comment (ethanfrey) [inline .circleci/config.yml]: wasmd-demo-demo ???

File: .circleci/config.yml [hunk 2]
```diff
@@ -4,11 +4,11 @@ executors:
   golang:
     docker:
       - image: circleci/golang:1.13
-    working_directory: /go/src/github.com/cosmwasm/wasmd
+    working_directory: /go/src/github.com/cosmwasm/wasmd-demo
```
Comment (ethanfrey) [inline .circleci/config.yml]: Why the rename here? This is for all tests

File: .circleci/config.yml (hunk 1 above)
Comment (alpe) [reply .circleci/config.yml]: ja, one demo to much 😆

File: .circleci/config.yml (hunk 2 above)
Comment (alpe) [reply .circleci/config.yml]: 😕 this was not intended.  should have marked this as draft.

**Pull request merged**